webcomponents/tool/src/main/java/org/sakaiproject/webcomponents/permissions/PermissionsEntityProvider.java

```diff
@@ -161,8 +173,22 @@ public String handleSet(EntityReference entityRef, Map<String, Object> params) {
 
         String reference = (String) params.get("ref");
 
+        AuthzGroup authzGroup;
+        try {
+            authzGroup = authzGroupService.getAuthzGroup(reference);
+        } catch (GroupNotDefinedException e) {
+            try {
+                authzGroup = authzGroupService.addAuthzGroup(reference);
```

Comment:
This doesn't seem a great way to do things, imo. Surely if content hosting relies on realms for its folder permissions, it should create them as the folders are created?

Reply:
custom permissions per folder is a 1% use case .... so if you created a new realm per folder, you would have hundreds of thousands of new realms that are unnecessary. and then you start introducing new edge case bugs: e.g., what happens when you adjust Resources permissions to remove content.read but all of your folders are still readable?

Reply:
Should the reference be checked that it is actually under the current site, if you know what I mean?

Reply:
yes i think prob a good addition. perms are checked later on and obviously logged .... but sure makes sense

Reply:
updated .....